### PR TITLE
add cached funct is_allowed

### PR DESCRIPTION
**Pull request by delcroip**

### CODE REVIEW

File: location/models.py [hunk 1]
```diff
@@ -147,6 +158,18 @@ def get_location_from_ids(self, qsr, loc_type):
         if loc_type:
             return [x for x in list(qsr) if x.type == loc_type]
         return list(qsr)
+    
+    def is_allowed(self, user, list_id):
```
Comment (weilu) [inline location/models.py:162]: s/list_id/location_id

File: location/models.py [hunk 2]
```diff
@@ -16,6 +18,15 @@
 
 logger = logging.getLogger(__file__)
 
+def free_cache_for_user(user_id):
+    cache_name = f'location_allowed_{user_id}'
```
Comment (weilu) [inline location/models.py:22]: Consider renaming this to `user_locations_{user_id}` to be consistent with the district cache name below.

File: location/models.py [hunk 3]
```diff
@@ -16,6 +18,15 @@
...
+    cache.delete(cache_name)
+    cache_name = f"user_districts_{user_id}"
+    cache.delete(cache_name)
+    
+@receiver(post_save, sender=core_models.InteractiveUser)
+def free_cache_post_user_save(sender, instance, created, **kwargs):
+    free_cache_for_user(instance.id)
```
Comment (weilu) [inline location/models.py:29]: Is it worth checking if `instance.officer.location` changed before invalidating the cache?

File: location/tests.py [hunk 4]
```diff
@@ -49,8 +49,9 @@ def test_children(self):
     def test_allowed_location(self):
         allowed = LocationManager().allowed(self.test_user.id, loc_types = ['R','D','W'], qs = True)
         self.assertEqual(len(allowed),3)
-
+        self.assertTrue(LocationManager().is_allowed(self.test_user, [self.test_village.parent.parent.id]), 'is_allowed function is not working as supposed')
         other = create_test_location('D',  custom_props={'parent':self.test_village.parent.parent.parent, 'code':'NOTALLO'})
         allowed = LocationManager().allowed(self.test_user.id, loc_types = ['R','D','W'],qs = True)
         self.assertEqual(len(allowed),2)
+        self.assertFalse(LocationManager().is_allowed(self.test_user, [other.id, self.test_village.parent.parent.id]), 'is_allowed function is not working as supposed')
```
Comment (weilu) [inline location/tests.py:56]: Should we add tests checking the state of the cache?

File: location/models.py (hunk 1 above)
Comment (delcroip) [reply location/models.py:162]: I put locations_id because it is a list

File: location/models.py (hunk 3 above)
Comment (delcroip) [reply location/models.py:29]: I don't want to slow the system, loading the is_allowed won't take so much time, the issue is how often it is called also user are not often modified

**Pull request merged**